lib/jsdom/living/geometry/DOMMatrixReadOnly-impl.js

```diff
@@ -0,0 +1,519 @@
+    const initializationArray = arrayFromDOMMatrixInit(matrixInitialization);
+
+    return new DOMMatrixReadOnlyImpl([initializationArray]);
+  }
+
+  static fromFloat32Array(array32) {
+    return new DOMMatrixReadOnlyImpl([Array.from(array32)]);
+  }
+
+  static fromFloat64Array(array64) {
+    return new DOMMatrixReadOnlyImpl([Array.from(array64)]);
+  }
+
+  translate(tX = 0, tY = 0, tZ = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.translateSelf(tX, tY, tZ);
+
+    return m;
+  }
+
+  scale(scaleX = 1, scaleY, scaleZ = 1, originX = 0, originY = 0, originZ = 0) {
+    if (typeof scaleY === "undefined") {
+      scaleY = scaleX;
+    }
+
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scaleSelf(scaleX, scaleY, scaleZ, originX, originY, originZ);
+
+    return m;
+  }
+
+  scaleNonUniform(scaleX = 1, scaleY = 1) {
+    // Deprecated: https://www.w3.org/TR/geometry-1/#dom-dommatrixreadonly-scalenonuniform
+
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scaleSelf(scaleX, scaleY);
+
+    return m;
+  }
+
+  scale3d(scale = 1, originX = 0, originY = 0, originZ = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.scale3dSelf(scale, originX, originY, originZ);
+
+    return m;
+  }
+
+  rotate(rotX = 0, rotY, rotZ) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.rotateSelf(rotX, rotY, rotZ);
+
+    return m;
+  }
+
+  rotateFromVector(x = 0, y = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.rotateFromVectorSelf(x, y);
+
+    return m;
+  }
+
+  rotateAxisAngle(x = 0, y = 0, z = 0, angle = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+    m.rotateAxisAngleSelf(x, y, z, angle);
+
+    return m;
+  }
+
+  skewX(sX = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.skewXSelf(sX);
+
+    return m;
+  }
+
+  skewY(sY = 0) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.skewYSelf(sY);
+
+    return m;
+  }
+
+  multiply(other) {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.multiplySelf(other);
+
+    return m;
+  }
+
+  flipX() {
+    const m = DOMMatrix.create([this._getCurrentMatrix()]);
+
+    m.multiplySelf(DOMMatrix.create([[-1, 0, 0, 1, 0, 0]]).toJSON());
+
+    return m;
+  }
+
+  flipY() {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.multiplySelf(DOMMatrix.createImpl([[1, 0, 0, -1, 0, 0]]).toJSON());
+
+    return m;
+  }
+
+  inverse() {
+    const m = DOMMatrix.createImpl([this._getCurrentMatrix()]);
+
+    m.invertSelf();
+
+    return m;
+  }
+
+  // transformPoint(point) {
+  //   throw new Error(`Transform Point is not implemented`);
+  // }
+
+  toFloat32Array() {
+    return new Float32Array(this._getCurrentMatrix());
+  }
+
+  toFloat64Array() {
+    return new Float64Array(this._getCurrentMatrix());
+  }
+
+  toJSON() {
```

Comment:
File a bug against https://github.com/jsdom/webidl2js to implement `[Default] toJSON()`, and link that here.